### PR TITLE
Wave2 server side support

### PR DESCRIPTION
**Pull request by akallabeth**
Automatically switches to `WAVE2 PDU` if client supports it.

### CODE REVIEW

File: channels/rdpsnd/server/rdpsnd_main.c [hunk 1]
```diff
@@ -65,8 +66,8 @@ UINT rdpsnd_server_send_formats(RdpsndServerContext* context, wStream* s)
 		Stream_Write_UINT16(s, context->server_formats[i].nChannels); /* nChannels */
 		Stream_Write_UINT32(s,
 		                    context->server_formats[i].nSamplesPerSec); /* nSamplesPerSec */
-		Stream_Write_UINT32(s, context->server_formats[i].nSamplesPerSec *
-		                    context->server_formats[i].nChannels *
+		Stream_Write_UINT32(s, context->server_formats[i].nSamplesPerSec*
+		                    context->server_formats[i].nChannels*
```
Comment (mfleisz) [inline channels/rdpsnd/server/rdpsnd_main.c]: imo the old formatting was more clear in this case ...

File: channels/audin/client/audin_main.c [hunk 2]
```diff
@@ -221,25 +220,33 @@ static UINT audin_process_formats(IWTSVirtualChannelCallback* pChannelCallback,
...
 		Stream_Read_UINT16(s, format.cbSize);
-		format.data = Stream_Pointer(s);
 
 		if (Stream_GetRemainingLength(s) < format.cbSize)
 			return ERROR_INVALID_DATA;
 
-		Stream_Seek(s, format.cbSize);
+		if (format.cbSize > 0)
+		{
+			Stream_Seek(s, format.cbSize);
```
Comment (hardening) [inline channels/audin/client/audin_main.c]: Don't we have a pb here, shouldn't you seek _after_ doing the copy ?

**Comment by akallabeth** [comment]
@hardening Thx, you spotted some nasty issue.

**Comment by akallabeth** [comment]
@hardening and the common part, I've added it before I realized we only need a common header.

**Pull request merged**